kvbc/include/merkle_builder/BaseMerkleBuilder.h

```diff
@@ -0,0 +1,76 @@
+// Concord
+//
+// Copyright (c) 2023 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the
+// LICENSE file.
+
+#pragma once
+
+#include "IMerkleBuilder.h"
+#include "kvbc_app_filter/kvbc_key_types.h"
+#include "categorization/base_types.h"
+
+namespace concord {
+namespace kvbc {
+namespace sparse_merkle {
+
+using Hasher = concord::kvbc::categorization::Hasher;
+
+class BaseMerkleBuilder : public IMerkleBuilder {
+ public:
+  bool needProcessing(char type) const {
+    if (type == kKvbKeyEthBalance || type == kKvbKeyEthCode || type == kKvbKeyEthStorage || type == kKvbKeyEthNonce) {
+      return true;
+    }
+    return false;
+  }
+
+  address getAddress(const std::string& key) const {
+    ConcordAssert(key.size() > IMerkleBuilder::address_size);  // Include 1 byte indicating key type
```

Comment:
You assume that it's an Ethereum deployment, it will crash if the key starts with with some need processing char but it's not Ethereum